mitreattack/navlayers/exporters/matrix_gen.py

```diff
@@ -195,9 +196,8 @@ def _get_technique_listing(self, tactic, domain='enterprise'):
         techs = self._search(domain, [Filter('type', '=', 'attack-pattern'),
                                       Filter('kill_chain_phases.phase_name', '=', tactic)])
         for entry in techs:
-            if entry['kill_chain_phases'][0]['kill_chain_name'] == 'mitre-attack' or \
-                            entry['kill_chain_phases'][0]['kill_chain_name'] == 'mitre-mobile-attack':
-                tid = [t['external_id'] for t in entry['external_references'] if 'attack' in t['source_name']]
+            if entry['kill_chain_phases'][0]['kill_chain_name'] in MITRE_ATTACK_ID_SOURCE_NAMES:
+                tid = [t['external_id'] for t in entry['external_references'][:1] if 'attack' in t['source_name']]
```

Comment:
sometimes `attack` is in the `source_name` field for non-ATT&CK objects, so that isn't the best way to get an ATT&CK ID. we probably need to find a better way to guarantee that this will always only get ATT&CK IDs.

Reply:
We will investigate ways to further optimize the ATT&CK ID detection process. For now, this PR should be merged as it resolves Issue #76 and still improves the aforementioned process.